### PR TITLE
Clear 'item' in Autocomplete widget when textbox cleared in IE

### PR DESCRIPTION
**Pull request by vansimke**
Internet Explorer has a "clear" button built into the native input HTML element. This button clears the value of the element, but didn't clear out the underlying value if it was a part of a widget that used the dijit/form/_AutocompleterMixin. This PR adds an event listener to the widget that will set the item to null when this button is pressed.

Fixes [#18862](https://bugs.dojotoolkit.org/ticket/18862)

### CODE REVIEW

File: form/_AutoCompleterMixin.js [hunk 1]
```diff
@@ -486,6 +486,16 @@ define([
...
+
+			// if the "clear" button is pressed in IE (provided by <input>), then clear the value
+			// must be async since the clearing of the input element happens after the mouseup event 
+			this.on("mouseup", function (e) {
+				setTimeout(lang.hitch(this, function() {
+					if (!e.target.value) {
+						this.set("item", null);
+					}
+				}), 0); 
+			});
```
Comment (wkeese) [inline form/_AutoCompleterMixin.js]: Can't you detect this via the oninput event instead?   That's supposed to be a catch-all event to include corner cases like this.

Comment (vansimke) [inline form/_AutoCompleterMixin.js]: Good idea, that would make this change more general. I've pushed changes to the code and tests to reflect this approach.

Comment (wkeese) [inline form/_AutoCompleterMixin.js]: OK, I'm glad that works.   My next question though is whether or not you can leverage `TextBoxMixin#_onInput()`.

There's a mess of code in TextBoxMixin.js for normalizing the input event, although it may all be useless now since IE8 and IE9 are effectively desupported by Microsoft... OTOH dojo and dijit never officially desupported them.

**Comment by wkeese** [comment]
Not sure if my last comment got lost or not.   Listening for the "input" event is an improvement but probably you should extend `TextBoxMixin#_onInput()` instead.

That's probably a bit better because here's a mess of code in TextBoxMixin.js for normalizing the input event, although it's probably only useful for <= IE9.

**Comment by vansimke** [comment]
Yeah, I lost track of the follow up on this. I'll take a look at it when I get a chance to get some more Dojo time in. Thanks for bringing this back to the top of my pile :)

**Comment by vansimke** [comment]
I did not intend to close this PR. I was merging in another branch and grabbed this by mistake. Still need to research @wkeese's idea.
